File: agents/windows-winui/Services/AppConfig.cs
using System.Text.Json.Serialization;

namespace LiveDashboardAgent.Services;

public sealed class AppConfig
{
    public string ServerUrl { get; init; } = "";
    public string Token { get; init; } = "";
    public int IntervalSeconds { get; init; } = 5;
    public int HeartbeatSeconds { get; init; } = 60;
    public int IdleThresholdSeconds { get; init; } = 300;
    public bool EnableLog { get; init; } = false;

    public static AppConfig Default { get; } = new();

    public AppConfig Normalize()
    {
        return new AppConfig
        {
            ServerUrl = ServerUrl.Trim(),
            Token = Token.Trim(),
            IntervalSeconds = Clamp(IntervalSeconds, 1, 300, 5),
            HeartbeatSeconds = Clamp(HeartbeatSeconds, 10, 600, 60),
            IdleThresholdSeconds = Clamp(IdleThresholdSeconds, 30, 3600, 300),
            EnableLog = EnableLog,
        };
    }

    public string? Validate()
    {
        var normalized = Normalize();
        if (string.IsNullOrWhiteSpace(normalized.ServerUrl))
        {
            return "服务器地址不能为空";
        }
        if (string.IsNullOrWhiteSpace(normalized.Token) || normalized.Token == "YOUR_TOKEN_HERE")
        {
            return "Token 不能为空";
        }
        if (!Uri.TryCreate(normalized.ServerUrl, UriKind.Absolute, out var uri) ||
            (uri.Scheme != Uri.UriSchemeHttps && uri.Scheme != Uri.UriSchemeHttp))
        {
            return "服务器地址必须使用 http:// 或 https://";
        }
        return null;
    }

    private static int Clamp(int value, int min, int max, int fallback)
    {
        return value < min || value > max ? fallback : value;
    }
}

internal sealed class LegacyJsonConfig
{
    [JsonPropertyName("server_url")]
    public string? ServerUrl { get; init; }

    [JsonPropertyName("token")]
    public string? Token { get; init; }

    [JsonPropertyName("interval_seconds")]
    public int? IntervalSeconds { get; init; }

    [JsonPropertyName("heartbeat_seconds")]
    public int? HeartbeatSeconds { get; init; }

    [JsonPropertyName("idle_threshold_seconds")]
    public int? IdleThresholdSeconds { get; init; }

    [JsonPropertyName("enable_log")]
    public bool? EnableLog { get; init; }

    public AppConfig ToAppConfig()
    {
        return new AppConfig
        {
            ServerUrl = ServerUrl ?? "",
            Token = Token ?? "",
            IntervalSeconds = IntervalSeconds ?? AppConfig.Default.IntervalSeconds,
            HeartbeatSeconds = HeartbeatSeconds ?? AppConfig.Default.HeartbeatSeconds,
            IdleThresholdSeconds = IdleThresholdSeconds ?? AppConfig.Default.IdleThresholdSeconds,
            EnableLog = EnableLog ?? AppConfig.Default.EnableLog,
        }.Normalize();
    }
}

File: agents/windows-flutter/lib/src/models/app_config.dart
class AppConfig {
  const AppConfig({
    required this.serverUrl,
    required this.token,
    required this.intervalSeconds,
    required this.heartbeatSeconds,
    required this.idleThresholdSeconds,
    required this.enableLog,
  });

  const AppConfig.empty()
    : serverUrl = '',
      token = '',
      intervalSeconds = 5,
      heartbeatSeconds = 60,
      idleThresholdSeconds = 300,
      enableLog = false;

  final String serverUrl;
  final String token;
  final int intervalSeconds;
  final int heartbeatSeconds;
  final int idleThresholdSeconds;
  final bool enableLog;

  AppConfig normalize() {
    return AppConfig(
      serverUrl: serverUrl.trim(),
      token: token.trim(),
      intervalSeconds: _clamp(intervalSeconds, 1, 300, 5),
      heartbeatSeconds: _clamp(heartbeatSeconds, 10, 600, 60),
      idleThresholdSeconds: _clamp(idleThresholdSeconds, 30, 3600, 300),
      enableLog: enableLog,
    );
  }

  String? validate() {
    final normalized = normalize();
    if (normalized.serverUrl.isEmpty) {
      return '服务器地址不能为空';
    }
    if (normalized.token.isEmpty || normalized.token == 'YOUR_TOKEN_HERE') {
      return 'Token 不能为空';
    }
    final uri = Uri.tryParse(normalized.serverUrl);
    if (uri == null ||
        !uri.hasScheme ||
        !(uri.scheme == 'http' || uri.scheme == 'https')) {
      return '服务器地址必须使用 http:// 或 https://';
    }
    return null;
  }

  AppConfig copyWith({
    String? serverUrl,
    String? token,
    int? intervalSeconds,
    int? heartbeatSeconds,
    int? idleThresholdSeconds,
    bool? enableLog,
  }) {
    return AppConfig(
      serverUrl: serverUrl ?? this.serverUrl,
      token: token ?? this.token,
      intervalSeconds: intervalSeconds ?? this.intervalSeconds,
      heartbeatSeconds: heartbeatSeconds ?? this.heartbeatSeconds,
      idleThresholdSeconds: idleThresholdSeconds ?? this.idleThresholdSeconds,
      enableLog: enableLog ?? this.enableLog,
    );
  }

  Map<String, Object?> toRegistryMap() {
    return {
      'server_url': serverUrl,
      'token': token,
      'interval_seconds': intervalSeconds,
      'heartbeat_seconds': heartbeatSeconds,
      'idle_threshold_seconds': idleThresholdSeconds,
      'enable_log': enableLog,
    };
  }

  static AppConfig fromMap(Map<Object?, Object?> map) {
    return AppConfig(
      serverUrl: (map['server_url'] as String?) ?? '',
      token: (map['token'] as String?) ?? '',
      intervalSeconds: _intValue(map['interval_seconds'], 5),
      heartbeatSeconds: _intValue(map['heartbeat_seconds'], 60),
      idleThresholdSeconds: _intValue(map['idle_threshold_seconds'], 300),
      enableLog: map['enable_log'] == true,
    ).normalize();
  }

  static int _clamp(int value, int min, int max, int fallback) {
    return value < min || value > max ? fallback : value;
  }

  static int _intValue(Object? value, int fallback) {
    if (value is int) {
      return value;
    }
    if (value is num) {
      return value.toInt();
    }
    if (value is String) {
      return int.tryParse(value) ?? fallback;
    }
    return fallback;
  }
}
